File: ORB.Core/Models/Benefit.cs
using System;
using System.Collections.Generic;

namespace ORB.Core.Models
{
    public class Benefit : ReportingSubEntity
    {
        public Benefit()
        {
            BenefitUpdates = new HashSet<BenefitUpdate>();
        }

        public int ProjectID { get; set; }
        public int? BenefitTypeID { get; set; }
        public int? MeasurementUnitID { get; set; }
        public decimal? TargetPerformanceLowerLimit { get; set; }
        public decimal? TargetPerformanceUpperLimit { get; set; }
        public int? RagOptionID { get; set; }
        public DateTime? BaselineDate { get; set; }
        public DateTime? ForecastDate { get; set; }
        public DateTime? ActualDate { get; set; }

        public virtual BenefitType BenefitType { get; set; }
        public virtual MeasurementUnit MeasurementUnit { get; set; }
        public virtual Project Project { get; set; }
        public virtual RagOption RagOption { get; set; }
        public virtual ICollection<BenefitUpdate> BenefitUpdates { get; set; }
    }
}

File: corporate-reporting/src/types/Benefit.ts
import { IEntity } from "./Entity";
import { IUser } from "./User";
import { IProject } from "./Project";
import { IReportingEntityWithDeliveryDates, ReportingEntityWithDeliveryDates } from "./ReportingEntity";
import { IBenefitUpdate } from "./BenefitUpdate";

export interface IBenefit extends IReportingEntityWithDeliveryDates {
    ProjectID: number;
    BenefitTypeID: number;
    MeasurementUnitID: number;
    TargetPerformanceUpperLimit: number | string;
    TargetPerformanceLowerLimit: number | string;
    LeadUserID: number;
    RagOptionID: number;
    BenefitType?: IEntity;
    LeadUser?: IUser;
    Project?: IProject;
    MeasurementUnit?: IEntity;
    Description: string;
    BenefitUpdates?: IBenefitUpdate[];
}

export class Benefit extends ReportingEntityWithDeliveryDates implements IBenefit {
    public ProjectID = null;
    public BenefitTypeID = null;
    public MeasurementUnitID = null;
    public TargetPerformanceLowerLimit = '';
    public TargetPerformanceUpperLimit = '';
    public LeadUserID = null;
    public RagOptionID = null;
    public ReportingFrequency = null;
    public Description = '';
}